tests/DistanceStraightTest.cpp

```diff
@@ -0,0 +1,289 @@
+/**
+ *  @file   DistanceStraightTest.cpp
+ *  @brief  DistanceStraightクラスのテスト
+ *  @author HaruArima08
+ */
+
+#include "DistanceStraight.h"
+#include <gtest/gtest.h>
+
+using namespace std;
+
+namespace etrobocon2025_test {
+
+  // 通常速度での直進テスト
+  TEST(DistanceStraightTest, RunNomalSpeed)
+  {
+    Robot robot;
+    // Power値の初期化
+    robot.getMotorControllerInstance().resetWheelsMotorPower();
+    double targetDistance = 1000.0;
+    double targetSpeed = 50.0;
+    double basePower = 100.0;
+    DistanceStraight ds(robot, targetDistance, targetSpeed);
+
+    // 初期値 （タイヤの累計走行距離）
+    int rightCount = robot.getMotorControllerInstance().getRightMotorCount();
+    int leftCount = robot.getMotorControllerInstance().getLeftMotorCount();
+    double leftInitial = Mileage::calculateWheelMileage(leftCount);
+    double rightInitial = Mileage::calculateWheelMileage(rightCount);
+
+    // 期待する走行距離
+    double expectedDistance = targetDistance;
+
+    // モーターに1回setMotorPower()を呼ぶと、モータカウントが power × 0.05分進む
+    // 1ステップ分の走行距離を許容誤差とする
+    double distanceError
+        = Mileage::calculateMileage(basePower * 0.05, basePower * 0.05);  // 許容誤差
+
+    double expectedError = 1;  // タイヤの走行距離の誤差の期待値
+
+    ds.run();  // 直進を実行
+
+    // 直進後の走行距離
+    rightCount = robot.getMotorControllerInstance().getRightMotorCount();
+    leftCount = robot.getMotorControllerInstance().getLeftMotorCount();
+    double actualDistance = Mileage::calculateMileage(rightCount, leftCount);
+
+    // 直進後のタイヤの累計走行距離
+    double rightActual = Mileage::calculateWheelMileage(rightCount);
+    double leftActual = Mileage::calculateWheelMileage(leftCount);
+    // タイヤごとの走行距離
+    double rightDifference = std::abs(rightActual - leftInitial);
+    double leftDifference = std::abs(leftActual - rightInitial);
+    // タイヤの走行距離の誤差
+    double actualError = std::abs(rightDifference - leftDifference);
+
+    // 走行距離のテスト
+    EXPECT_NEAR(expectedDistance, actualDistance, distanceError);
+    // 直進できているかのテスト
+    EXPECT_LE(actualError, expectedError);
+  }
+
+  // 最大速度での直進テスト
+  TEST(DistanceStraightTest, RunFullPower)
+  {
+    Robot robot;
+    // Power値の初期化
+    robot.getMotorControllerInstance().resetWheelsMotorPower();
+    double targetDistance = 1000.0;
+    double targetSpeed = 100.0;
```

Comment:
最大速度ならもっと大きそう．あと，最大がどれかわからないから，「十分大きい速度でのテスト」とかにする？
```suggestion
    double targetSpeed = 1000.0;
```